src/index.ts

```diff
@@ -1,10 +1,12 @@
 import type * as Babel from "@babel/core";
 import type { types as t, NodePath } from "@babel/core";
-import { createRequire } from "node:module";
 import { ImportUtil } from "babel-import-util";
 import { globalId } from "./global-id.ts";
-const req = createRequire(import.meta.url);
-const { default: decoratorSyntax } = req("@babel/plugin-syntax-decorators");
+
+// @ts-expect-error no upstream types
+import PluginSyntaxDecorators from "@babel/plugin-syntax-decorators";
+const decoratorSyntax =
+  PluginSyntaxDecorators.default || PluginSyntaxDecorators;
```

Comment:
I am loathe to spread this kind of incorrect code further through the ecosystem.

I think this is a bug in tsup: https://github.com/egoist/tsup/issues/1078

We don't need to use tsup, I'm fine with using babel or tsc directly instead.